### PR TITLE
feat: Add Nemotron-Labs-Diffusion (hybrid) SFT support

### PR DESCRIPTION
**Pull request by zyzhou5**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog

- Add specific line by line info of high level changes in this PR.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: docs/guides/dllm/finetune.md [hunk 1]
```diff
@@ -62,6 +64,7 @@ The recipe uses a **strategy pattern** to handle differences between model famil
 | Mode | Strategy | Description |
 |------|----------|-------------|
 | `mdlm` | `MDLMStrategy` | LLaDA-style: model receives corrupted tokens, MDLM cross-entropy loss |
+| `hybrid` | `HybridStrategy` | Nemotron-Labs-Diffusion-style: model receives clean tokens + `masked_indices`, combined diffusion + AR loss |
 
 ### LLaDA Configuration
```
Comment (jgerh) [inline docs/guides/dllm/finetune.md:69]: Changed procedural title to an imperative verb

```suggestion
### Configure LLaDA
```

File: docs/guides/dllm/finetune.md [hunk 2]
```diff
@@ -82,23 +85,59 @@ dataset:
   unshifted: true             # Required for dLLM training
 ```
 
+### Nemotron-Labs-Diffusion Configuration
```
Comment (jgerh) [inline docs/guides/dllm/finetune.md:88]:
```suggestion
### Configure Nemotron-Labs-Diffusion
```

File: docs/guides/dllm/finetune.md [hunk 3]
```diff
@@ -82,23 +85,59 @@ dataset:
...
 ### Key dLLM Config Fields
 
 | Field | Description |
 |-------|-------------|
-| `dllm.mode` | Training strategy (`mdlm`) |
-| `dllm.mask_token_id` | Token ID used for masking (`126336` for LLaDA) |
+| `dllm.mode` | Training strategy (`mdlm` or `hybrid`) |
+| `dllm.mask_token_id` | Token ID used for masking (`126336` for LLaDA, `100` for Nemotron-Labs-Diffusion) |
 | `dllm.eps` | Minimum corruption ratio to avoid zero-corruption samples |
+| `dllm.block_size` | When set, use block-wise corruption (otherwise uniform). Hybrid mode only. |
```
Comment (jgerh) [inline docs/guides/dllm/finetune.md:118]:
```suggestion
| `dllm.block_size` | Block-wise corruption size; uses uniform corruption when unset. Hybrid mode only. |
```

File: docs/guides/dllm/finetune.md
Comment (jgerh) [inline docs/guides/dllm/finetune.md:147]:
```suggestion
### Generate with LLaDA
```

Comment (jgerh) [inline docs/guides/dllm/finetune.md:31]:
```suggestion
| **4. Generate** | [Generate Text](#generate-text) | Generate text from a fine-tuned checkpoint |
```

File: docs/guides/dllm/finetune.md [hunk 4]
```diff
@@ -82,23 +85,59 @@ dataset:
...
 
+Nemotron-Labs-Diffusion (uses TP=2 by default):
+
+```bash
+torchrun --nproc-per-node=8 \
+    nemo_automodel/recipes/dllm/train_ft.py \
+    -c examples/dllm_sft/nemotron_labs_diffusion_sft.yaml
+```
+
 ## Generation / Inference
```
Comment (jgerh) [inline docs/guides/dllm/finetune.md:141]: Pls verify suggested changes: Updated the heading to use an imperative verb and to match the argparse CLI description for generating text from LLaDA checkpoints. Also changes line R31. 

```suggestion
## Generate Text
```

**Review by jgerh — COMMENTED** [review]
Completed tech pubs review of docs/guides/dllm/finetune.md and provided a few suggested text revisions.

**Review by pthombre — DISMISSED** [review]

**Review by pthombre — DISMISSED** [review]

**Review by pthombre — APPROVED** [review]
